server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -131,7 +132,18 @@ public ContextIndexSearcher(
         int maximumNumberOfSlices,
         int minimumDocsPerSlice
     ) throws IOException {
-        super(wrapWithExitableDirectoryReader ? new ExitableDirectoryReader((DirectoryReader) reader, cancellable) : reader, executor);
+        super(
+            wrapWithExitableDirectoryReader ? new ExitableDirectoryReader((DirectoryReader) reader, cancellable) : reader,
+            executor == null ? null : r -> {
+                try {
+                    executor.execute(r);
+                } catch (RejectedExecutionException ignored) {
+                    // in case of rejection run on the current thread without forking, we don't want failures in the Lucene search in the
+                    // unlikely case of running into a full queue on the executor side
+                    r.run();
```

Comment:
this looks good, I wonder if it should be the default behaviour now in TaskExecutor as well.

Reply:
Hmm that's a good question. Not sure, we do this purposely now but if we do it in Lucene by default, we're pretty much quietly robbing other users of a signal?

Reply:
Yea it's debatable, the thing is that if we now can rely on a single executor, and we have one thread executing the main search there as well as pulling tasks from the shared queue, it sounds natural to also ignore rejections and execute in the thread that is pulling tasks anyways? What would we do otherwise? Would we interrupt the whole search based on a task rejection and cancel following tasks?

Reply:
By the way, I am suggesting a potential improvement to Lucene but it makes sense to merge this to ES to start with.

Reply:
Actually we merged this in Lucene now in https://github.com/apache/lucene/pull/13622 so I dropped it here :)

Reply:
++ thanks!